faiss.py

```diff
@@ -0,0 +1,54 @@
+
+import pickle
+import faiss
+import abc
+
+
+
+class Indexer(abc.ABC):
+
+  @abc.abstractmethod
+  def __init__(self, dim, param, nprobe):
+    pass
+
+ 
+  @abc.abstractmethod
+  def add(self, vs):
+    pass
+  
+  @abc.abstractmethod
+  def train(self, train_vectors):
+    pass
+
+  @abc.abstractmethod
+  def find(self, query, topn): 
+    pass
+
+  def save(self,index):
```

Comment:
Implementations of save and load methods you should move too

Reply:
Done!